### ISSUE TITLE
Document a gotcha with some focus trap libraries and Shadow DOM

### ISSUE THREAD

**Issue by wkillerud** [warp-ds/elements#460]
@joms and I got hit by a focus trap in a dependency that was not configured correctly to deal with shadow DOM. The issue we saw was that the Slider component rendered fine, but no interactive element inside the component was clickable.

The component was being rendered in a bottom sheet on mobile. Outside that bottom sheet the component worked fine, also with emulated touch events.

Turns out the bottom sheet component used `focus-trap`. The `focus-trap` dependency only supports focusable elements inside a shadow root when opted in ( 🤷 ). So what was happening was the focus trap thought clicks inside the shadow root were outside the trap and should be cancelled.

https://github.com/stipsan/react-spring-bottom-sheet/blob/3cb57d11c9a1e3a23e4e855e9525abc380b527b2/src/hooks/useFocusTrap.tsx#L32

We patched in the option `tabbableOptions:{getShadowRoot:true}` using `pnpm patch` in the project and suddenly things started working again.

This might be a problem with other focus trap libraries. We should document this as a gotcha. I don't think it's limited to React.
